radar-commons-android/src/main/java/org/radarcns/android/auth/portal/MetaTokenParser.java

```diff
@@ -34,6 +35,7 @@ public AppAuthState parse(@NonNull String authString) throws IOException {
                     .property(MP_REFRESH_TOKEN_PROPERTY, json.getString("refreshToken"))
                     .property(PRIVACY_POLICY_URL_PROPERTY, json.getString("privacyPolicyUrl"))
                     .property(BASE_URL_PROPERTY, json.getString("baseUrl"))
+                    .property(CLIENT_SECRET_PROPERTY, json.getString("clientSecret"))
```

Comment:
This should probably not go in then. There should just be a check whether the secret matches.